com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/ThreadsRequestHandler.java

```diff
@@ -90,7 +110,10 @@
     private CompletableFuture<Response> stepIn(Requests.StepInArguments arguments, Response response, IDebugAdapterContext context) {
         ThreadReference thread = DebugUtility.getThread(context.getDebugSession(), arguments.threadId);
         if (thread != null) {
-            DebugUtility.stepInto(thread, context.getDebugSession().getEventHub());
+            setPendingStepKind(thread, StepRequest.STEP_INTO);
```

Comment:
There is already use StepRequest.STEP_INOT/STEP_OVER in the DebugUtility. I would suggest put the same logic either on inside DebugUtility.stepxxx or all on here. But not on different method call stacks.

Reply:
This logic is to record the thread's stepping state, and putting the code in ThreadRequestHandler is reasonable because ThreadRequestHandler life cycle is debug session dependent.